kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,901 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "processes/from_json_check_result_process.h"
+
+namespace Kratos
+{
+/// Local Flags
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, CORRECT_RESULT,                 0 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, HISTORICAL_VALUE,               1 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, USE_NODE_COORDINATES,           2 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, CHECK_ONLY_LOCAL_ENTITIES,      3 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, NODES_CONTAINER_INITIALIZED,    4 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, ELEMENTS_CONTAINER_INITIALIZED, 5 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, NODES_DATABASE_INITIALIZED,     6 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, ELEMENTS_DATABASE_INITIALIZED,  7 );
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+FromJSONCheckResultProcess::FromJSONCheckResultProcess(
+    Model& rModel,
+    Parameters ThisParameters
+    ) : mrModelPart([](Model& rModel,const Parameters& rParameters)->ModelPart&{const std::string& r_model_part_name = rParameters["model_part_name"].GetString(); const std::string sub_model_part_name = rParameters.Has("sub_model_part_name") ? rParameters["sub_model_part_name"].GetString() : ""; if (sub_model_part_name == "") { return rModel.GetModelPart(r_model_part_name); } else { return rModel.GetModelPart(r_model_part_name).GetSubModelPart(sub_model_part_name);}}(rModel, ThisParameters)),
+        mThisParameters(ThisParameters)
+{
+    mThisParameters.ValidateAndAssignDefaults(this->GetDefaultParameters());
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+FromJSONCheckResultProcess::FromJSONCheckResultProcess(
+    ModelPart& rModelPart,
+    Parameters ThisParameters
+    ) : mrModelPart(rModelPart),
+        mThisParameters(ThisParameters)
+{
+    mThisParameters.ValidateAndAssignDefaults(this->GetDefaultParameters());
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FromJSONCheckResultProcess::Execute()
+{
+    KRATOS_TRY;
+
+    ExecuteInitialize();
+    ExecuteFinalizeSolutionStep();
+    ExecuteFinalize();
+
+    KRATOS_CATCH("");
+}
```

Comment:
```suggestion
```

Why is this method needed? I really don't like processes that both have all the methods AND the `Execute` method, this is really confusing to use and error-prone (e.g. in this case you miss the call to `Check`!